src/zh/toptoon/src/eu/kanade/tachiyomi/extension/zh/toptoon/Toptoon.kt

```diff
@@ -0,0 +1,193 @@
+
+class Toptoon : ParsedHttpSource() {
+    override val name: String = "Toptoon頂通"
+    override val lang: String = "zh"
+    override val supportsLatest = true
+    override val baseUrl = "https://www.toptoon.net"
+    private val json: Json by injectLazy()
+
+    // Popular
+
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/ranking", headers)
+    override fun popularMangaNextPageSelector() = throw UnsupportedOperationException()
+    override fun popularMangaSelector() = throw UnsupportedOperationException()
+    override fun popularMangaFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val jsonUrl = response.body.string()
+            .substringAfter("jsonFileUrl: [\"")
+            .substringBefore("\"")
+            .replace("\\/", "/")
+        val jsonResponse = client.newCall(GET("https:$jsonUrl", headers)).execute()
+        val mangas = jsonResponse.parseAs<PopularResponseDto>().adult.map {
+            SManga.create().apply {
+                title = it.meta.title
+                url = it.url
+                thumbnail_url = it.thumbnail.url
+            }
+        }
+        return MangasPage(mangas, false)
+    }
+
+    // Latest
+
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/search", headers)
+    override fun latestUpdatesNextPageSelector() = throw UnsupportedOperationException()
+    override fun latestUpdatesSelector() = throw UnsupportedOperationException()
+    override fun latestUpdatesFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val jsonUrl = response.body.string()
+            .substringAfter("var jsonFileUrl = '")
+            .substringBefore("'")
+        val jsonResponse = client.newCall(GET("https:$jsonUrl", headers)).execute()
+        val mangas = jsonResponse.parseAs<Map<String, MangaDto>>().values
+            .sortedByDescending { it.lastUpdated.pubDate }
+            .map {
+                SManga.create().apply {
+                    title = it.meta.title
+                    url = it.url
+                    thumbnail_url = it.thumbnail.url
+                }
+            }
+        return MangasPage(mangas, false)
+    }
+
+    // Search
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) = GET("$baseUrl/search", headers)
+    override fun searchMangaNextPageSelector() = throw UnsupportedOperationException()
+    override fun searchMangaSelector() = throw UnsupportedOperationException()
+    override fun searchMangaFromElement(element: Element) = throw UnsupportedOperationException()
+
+    // Copied from Mihon, I only change parameter of searchMangaParse
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        return Observable.defer {
```

Comment:
Should do

```kotlin
client.newCall(searchMangaRequest(page, query, filters)).asObservableSuccess()
    .map { response -> searchMangaParse(response, query) }
```

You can also use `$baseUrl/search#$query` and get the query by `response.request.url.fragment!!` to avoid the need to .override `fetchSearchManga`.